File: apps/ihg/src/server/entity/Sport.ts
import { Schema, model, models } from "mongoose";

const SportSchema = new Schema({
  name: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  imgUrl: {
    type: String,
    required: true,
  },
  standings: [
    {
      hall: {
        type: String,
        required: true,
      },
      point: {
        type: Number,
        required: true,
        min: 1,
      },
      position: {
        type: Number,
        required: true,
        min: 1,
      },
    },
  ],
});

const Sport = models.Sport || model("Sport", SportSchema);
export default Sport;

File: pkg/models/sport.go
package models

import "go.mongodb.org/mongo-driver/v2/bson"

type SportStanding struct {
	Hall     string `bson:"hall" json:"hall"`
	Point    int    `bson:"point" json:"point"`
	Position int    `bson:"position" json:"position"`
}

type Sport struct {
	ID          bson.ObjectID   `bson:"_id,omitempty" json:"_id,omitempty"`
	Name        string          `bson:"name" json:"name"`
	Description string          `bson:"description" json:"description"`
	ImgUrl      string          `bson:"imgUrl" json:"imgUrl"`
	Standings   []SportStanding `bson:"standings" json:"standings"`
}
